autopts/client.py

```diff
@@ -1373,12 +1373,15 @@ def sigint_handler(sig, frame):
             signal.signal(signal.SIGINT, sigint_handler)
 
             return self.main(args)
+        except KeyboardInterrupt:
+            # Handling Ctrl-C
+            pass
+        except SystemExit as e:
+            if e.code != 0:
+                logging.exception(e)
         except BaseException as e:
-            if not isinstance(e, KeyboardInterrupt):   # Ctrl-C
-                if e.code != 0:
-                    # Exit with traceback
-                    logging.exception(e)
-
+            logging.exception(e)
```

Comment:
I think handling `BaseException` as a last effort is OK (but not ideal). We should try to cover all known exceptions specifically, so that the code becomes more obvious. 

I would thus still suggest to add
```
except TimeoutError as e:
    logging.exception(e)
```
before `except BaseException`

or do
```
except (TimeoutError, BaseException) as e:
    logging.exception(e)
```

but the best approach would be to not catch `BaseException` (or `Exception`) at all IMO, which is also the recommendation by PEP8. 

Since it's advised against in PEP8, most Python linters will also complain about a `except BaseException` or `except BaseException`

@sjanc should we consider following PEP8 and/or adding a style guide for autopts, and use a linter in CI?

Reply:
yeap, I think enforcing PEP8 is good thing (maybe with some exceptions, line allow lines longer than 79. eg 120)

there is some cleanup to be done before this can be enabled on default though :)

Reply:
Indeed. Refactoring the entire codebase will take a while :D 

https://github.com/auto-pts/auto-pts/issues/1338